src/internal/storage/kv/lru_cache.go

```diff
@@ -0,0 +1,111 @@
+package kv
+
+import (
+	"context"
+	"sync"
+
+	"github.com/hashicorp/golang-lru/v2/simplelru"
+	"github.com/pachyderm/pachyderm/v2/src/internal/log"
+	"github.com/pachyderm/pachyderm/v2/src/internal/stream"
+	"go.uber.org/zap"
+)
+
+var _ Store = &LRUCache{}
+
+type LRUCache struct {
+	slow, fast Store
+
+	mu      sync.RWMutex
+	cache   simplelru.LRU[string, struct{}]
```

Comment:
I'm a little confused on the purpose of `cache` here. I see we already have a `slow` and `fast` store. The LRU process when applied to slow and fast makes sense. But it looks like we are also updating the internal cache when we do a put and delete. I assume that the simple lru cache handles updating the eviction logic for us, but if we have cache, why do we need fast?